pkg/utils/utils.go

```diff
@@ -418,40 +417,15 @@ func EnvVarResourceFieldSelectorEqual(resource1, resource2 v1.ResourceFieldSelec
 		resource1.Divisor.Cmp(resource2.Divisor) == 0
 }
 
-func SetProxyEnvVars(proxyConfig *configv1.Proxy) []v1.EnvVar {
+func GetProxyEnvVars() []v1.EnvVar {
 	envVars := []v1.EnvVar{}
-	if proxyConfig == nil {
-		return envVars
-	}
-	if len(proxyConfig.Status.HTTPSProxy) != 0 {
-		envVars = append(envVars, v1.EnvVar{
-			Name:  "HTTPS_PROXY",
-			Value: proxyConfig.Status.HTTPSProxy,
-		},
-			v1.EnvVar{
-				Name:  "https_proxy",
-				Value: proxyConfig.Status.HTTPSProxy,
-			})
-	}
-	if len(proxyConfig.Status.HTTPProxy) != 0 {
-		envVars = append(envVars, v1.EnvVar{
-			Name:  "HTTP_PROXY",
-			Value: proxyConfig.Status.HTTPProxy,
-		},
-			v1.EnvVar{
-				Name:  "http_proxy",
-				Value: proxyConfig.Status.HTTPProxy,
-			})
-	}
-	if len(proxyConfig.Status.NoProxy) != 0 {
-		envVars = append(envVars, v1.EnvVar{
-			Name:  "NO_PROXY",
-			Value: proxyConfig.Status.NoProxy,
-		},
-			v1.EnvVar{
-				Name:  "no_proxy",
-				Value: proxyConfig.Status.NoProxy,
+	for _, envvar := range []string{"HTTPS_PROXY", "https_proxy", "HTTP_PROXY", "http_proxy", "NO_PROXY", "no_proxy"} {
+		if value := os.Getenv(envvar); value != "" {
+			envVars = append(envVars, v1.EnvVar{
+				Name:  envvar,
+				Value: value,
```

Comment:
just to clarify (and maybe you want to add a comment so its clear why this is working) we're checking for these env vars on the CLO because OLM will add it to the operator deployment, right?

Reply:
> just to clarify (and maybe you want to add a comment so its clear why this is working) we're checking for these env vars on the CLO because OLM will add it to the operator deployment, right?

This was the message as I understood from OLM team, OLM will inject those vars into the operator deployments